### PR TITLE
Private Events handler

### PR DESCRIPTION
**Pull request by RokoVarano**
This site is about creating events as a user, so that other users can sing up for them.

Anonymous users can:
  - Click an event to see it's description and who is going
  - Log in
  - Create a new account

Logged in users can:
  - Create new Events
  - Subscribe to other events by clicking the Attend button

Built with

- Ruby 3.0.0 
- Rails 6
- HTML5
- Bootstrap (CDN)

### CODE REVIEW

**Review by peterrobert — CHANGES_REQUESTED** [review]
### INVALID 🛑 
Hello, @RokoVarano  and @JohnHernCode Great work submitting your project for a review. This shows great effort 👏🏿 . However, for your project to be reviewed, it needs to meet the minimum requirements. Kindly check below ⬇️ 

- [x] Kindly add linters to your project. This will make your project more professional and also allow you to follow best practices. Kindly add linters to your project and submit your project for another review.

linter_config

### if you have any question kindly reach me via slack @peterrobert

### HAPPY CODING 💻

**Review by NtwaliHeritier — CHANGES_REQUESTED** [review]
### **STATUS: CHANGES REQUESTED🛑🛑🛑🛑**

Hello team,
This is an outstanding and great project. I can tell you put in a lot of effort.
Let's make some small changes and make it more impeccable.

[REQUIRED]: In the events_controller's index action, kindly add two lines for upcoming events and past events. Then use the scope you defined in the event model for each of them.

[REQUIRED]: In the users_controller's show action, add @created_events = current_user.events, to show events created by the logged in user. Also add variable for past and upcoming events to be attended by the current user.

[REQUIRED]: Kindly add unit and integration tests to the project.

That's it my friends.
Implement the required changes and ask for another review👍👌
Happy coding💻💻

**Review by mahmoud717 — CHANGES_REQUESTED** [review]
### Changes required ♻️ 

**Hello guys**

Great job with the project so far it looks amazing, but there are still some changes to be made:
 
---
### Requested 🔴 

- [x] you should make it that a user can't attend the same event twice.
- [x] should add more tests, right now your app only has one test. it should have more unit and integration tests.



---

please make these changes and request another code review.

if you have any questions don't hesitate to contact me on slack **MahmoudMohammad**

File: README.md [hunk 1]
```diff
@@ -1,25 +1,77 @@
...
 
-* Services (job queues, cache servers, search engines, etc.)
+To get a local copy up and running follow these simple example steps.
 
-* Deployment instructions
+### Install
+- Clone the [repo](https://github.com/JohnHernCode/private-ev)
+- In the terminal, cd 'private-ev'
+- run 'bundle install'
+- run 'rails db:migrate'
```
Comment (acushlakoncept) [inline README.md]: Let's make our Install instruction look a bit professional. Enclose commands with backtick(`). Example in your case (don't include the hash (#))
```
#```
git clone https://github.com/JohnHernCode/private-ev
cd private-ev
bundle install
rails db:migrate
#```
```
Make sure to include `rails db:create` before `rails db:migrate` and `yarn`

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -8,9 +8,22 @@
...
+      <p class='alert'><%= alert %> </p>
+    </div>
+    <% elsif flash.now[:notice] %>
+    <div class='notification is-info global-notification'>
+      <p class='notice'><%= notice %> </p>
+    </div>
+    <%else%>
+    <%end%>
+    <%= render :partial => 'layouts/navbar' %>
     <%= yield %>
```
Comment (acushlakoncept) [inline app/views/layouts/application.html.erb]: There should be absolutely NO logic in views.

File: app/views/layouts/_navbar.html.erb [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+  
+  <% if logged_in? %>
+  <span style="padding-right: 10px;">Hi! "<%= current_user.username.upcase %>"</span>
+  <%= button_to "Logout", '/login', method: "delete" %>
+  <%= button_to "Create", '/event', method: :get %>
+  <%= button_to "My Events", '/show', method: :get %>
+  <%else%>
+  <%= button_to "Login", '/login', method: :get %>
+  <%= button_to "Sign Up", new_user_path, method: :get %>
+  <%end%>
```
Comment (acushlakoncept) [inline app/views/layouts/_navbar.html.erb]: There should be absolutely NO logic in views.

File: app/views/events/show.html.erb [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+<br/>
+
+<% if @event.attendees.empty?%>
+<% else %>
+
+<h6>Who's Coming?</h6>
+<ul>
+<% @event.attendees.each do |person|%>
+  <li><%= person.username %></li>
+<%end%>
```
Comment (acushlakoncept) [inline app/views/events/show.html.erb]: There should be absolutely NO logic in views.

**Review by acushlakoncept — CHANGES_REQUESTED** [review]
# Status: Changes Required ♻️

### Hello @JohnHernCode  and @RokoVarano, 

You've done an awesome job so far 👌. But there are still some changes that are necessary for this project to be approved.

- [ ] Don't you think there is no need for the owner of the event to have to click the **Attend** Event button since he/she is the **host** already

- [x] You should remove the attend button from past event as there is no need to attend an event that has already past.

- [x] The user's show page should display the events the user has created 👍  and events that the user is attending to separated in past and upcoming ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))

- [x] Add how to run tests to your ReadMe file.

- [x] Your specs files looks good with a wide tests coverage, but for some reason when I run `rspec` only one example runs. Kindly fix this.
Hint: I noticed running each spec went well, say `rspec spec/features/users_spec.rb` except `rspec spec/features/events_spec.rb` and removing it to run `rspec` ran all the examples, so you probably have an issue with your `feature/events_spec.rb` file.

- [x] There should be absolutely NO logic in views. Views are only for displaying information that became available to it from its corresponding controller not generating it.  ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/section/no_logic_in_helpers_requirement.png) from Pathwright)
I have gone ahead and made some comments on your codebase see affected files below:

**Don't know where to start or how to?** See  [Rails View Helpers Best Practices](https://medium.com/@acushla4real/rails-view-helpers-best-practices-guide-d7d7b6db3db4) 

### Cheers and Happy coding! 💻 👏

(Feel free to contact me on slack(@acushla) or reply to this thread, if you have any additional question)
Uduak Essien
[Twitter](https://twitter.com/acushlakoncept) | [LinkedIn](https://www.linkedin.com/in/acushlakoncept/)

**Review by umairarshadbutt — CHANGES_REQUESTED** [review]
# Status: REQUIRED CHANGES :warning: :construction: 

Hi @JohnHernCode and @RokoVarano 
Good job with the changes! :tada: We are almost there.
We, however, do have a few changes to carry out before we proceed.

##

Kindly work on the suggested changes 👇.
- [x] there is one test case that is failing, please fix it.
##
- **When you have made the specified changes, please submit another code review. Thank you!**
##
**If you are stuck at any point or have any questions, let me know by leaving a comment here and mentioning me. I will be happy to help**
HAPPY CODING :computer:
**[Umair Arshad](https://umairarshad.me/)**
[Twitter](https://twitter.com/its_UmairArshad) | [LinkedIn](https://www.linkedin.com/in/umair-arshad-butt/) | [GitHub](https://github.com/umairarshadbutt)

**Review by oloomoses — APPROVED** [review]
## Congratulations, Project Approved. :muscle: :tada: :clap: 
**Great job working on the suggested changes from the previous review.
You have passed all the requirements for this project :muscle: 
Feel free to merge and close this pull request.**

**I wish you the best in your next project and see you at the top :rocket:** 
#### Happy Coding
_Slack: Oloo Moses_

**Pull request merged**